### PR TITLE
RXR-2293 add wrapper function to run trial more easily

### PR DESCRIPTION
**Pull request by roninsightrx**
- Wraps the loop-over-patients into a function
- Introduces a new `create_model_design` feature
- Introduces a new `create_initial_dose_design` feature
- allows to specify the design using a YAML file (optional)
- Adds a few things to the output from the simulation (est / true parameters / GOF from MAP fits / final AUC estimates)
- Adds test for new functions
- Adds a vignette that showcases the functionality

### CODE REVIEW

File: R/sample_and_adjust.R [hunk 1]
```diff
@@ -32,7 +32,7 @@
 #' @param verbose verbose output?
 #' @returns a named list containing `final_regimen` (all doses after
 #' adjustment), `tdms` (all collected levels, both true and measured, that is,
-#' both with and without residual variability), and `additional_info`, which
+#' both with and without residual variability), and `est_parameters`, which
```
Comment (roninsightrx) [inline R/sample_and_adjust.R:35]: renaming this output variable, as it was just holding the fitted parameters.

File: inst/md/example1.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    anchor: dose
+  regimen_update:
+    at: [3, 5, 7]
+    anchor: dose
+    update_type: dose
+    dose_optimization_method: map_adjust_dose
+  sim:
+    lib: pkvancouvm
+  est:
+    lib: pkvancouvm
```
Comment (jasmineirx) [inline inst/md/example1.yaml:23]: models referenced in this package should be available in PKPDsim (https://github.com/InsightRX/PKPDsim/tree/master/inst/models)

File: R/sample_and_adjust.R (hunk 1 above)
Comment (jasmineirx) [reply R/sample_and_adjust.R:35]: in another PR we add other info here for other fit methods. i would prefer to keep this generic.

Comment (roninsightrx) [reply R/sample_and_adjust.R:35]: ah didn't see that. I can move it back in. My thinking was that "additional_info" wasn't really something where a user would think to look for parameters, so created a separate object to collect tghat. I would still leave it separate. IMO we should leave `additional_info` to collect only more miscellaneous stuff.

**Review by jasmineirx — APPROVED** [review]
made a few more fixes to bring everything together! looks good to me. It would be nice to add a test for run_trial, although the vignettes do a pretty good job catching errors. It would also be better to put files used only for testing in `tests/testthat/test` instead of as an installed file.

**Pull request merged**